sunbeam-python/sunbeam/jobs/manifest.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import yaml
+from pydantic import Field, ValidationError
+from pydantic.dataclasses import dataclass
+
+from sunbeam.clusterd.client import Client as clusterClient
+from sunbeam.jobs.common import BaseStep, Result, ResultType, Status
+
+LOG = logging.getLogger(__name__)
+EMPTY_MANIFEST = """charms: {}
+terraform-plans: {}
+"""
+
+
+@dataclass
+class JujuManifest:
+    bootstrap_args: List[str] = Field(
+        alias="bootstrap-args", description="Extra args for juju bootstrap"
+    )
+
+
+@dataclass
+class CharmsManifest:
+    channel: Optional[str] = Field(default=None, description="Channel for the charm")
+    revision: Optional[int] = Field(
+        default=None, description="Revision number of the charm"
+    )
+    rocks: Optional[Dict[str, str]] = Field(
+        default=None, description="Rock images for the charm"
+    )
+    config: Optional[Dict[str, Any]] = Field(
+        default=None, description="Config options of the charm"
+    )
+    source: Optional[Path] = Field(default=None, description="Local charm bundle path")
+
+
+@dataclass
+class TerraformManifest:
+    source: Path = Field(description="Path to Terraform plan")
+
+
+@dataclass
+class Manifest:
+    juju: Optional[JujuManifest] = None
+    charms: Optional[Dict[str, CharmsManifest]] = None
+    terraform_plans: Optional[Dict[str, TerraformManifest]] = Field(
+        default=None, alias="terraform-plans"
+    )
+
+    @classmethod
+    def load(cls, manifest_file: Path) -> "Manifest":
+        try:
+            with manifest_file.open() as file:
+                return Manifest(**yaml.safe_load(file))
+        except FileNotFoundError as e:
+            raise e
+        except ValidationError as e:
+            raise e
```

Comment:
There's no value in just re-raising exceptions, you could either get rid of the whole try, or do something about exceptions

Reply:
Removed exceptions